src/hooks/popupform.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useCallback } from 'react';
+
+const popupForm = () => {
```

Comment:
`popupForm` should be named `useFormAndValidation` because any custom hook should be named with `use` in order to let React know it's a custom hook. And you don't need to have `popup` in the name because you can control any form (not inside popups)